src/main/java/net/minecraftforge/common/ForgeMod.java

```diff
@@ -226,48 +226,50 @@ public void missingSoundMapping(RegistryEvent.MissingMappings<SoundEvent> event)
+    public void registerRecipeSerializers(RegisterEvent event)
     {
-        CraftingHelper.register(AndCondition.Serializer.INSTANCE);
-        CraftingHelper.register(FalseCondition.Serializer.INSTANCE);
-        CraftingHelper.register(ItemExistsCondition.Serializer.INSTANCE);
-        CraftingHelper.register(ModLoadedCondition.Serializer.INSTANCE);
-        CraftingHelper.register(NotCondition.Serializer.INSTANCE);
-        CraftingHelper.register(OrCondition.Serializer.INSTANCE);
-        CraftingHelper.register(TrueCondition.Serializer.INSTANCE);
-        CraftingHelper.register(TagEmptyCondition.Serializer.INSTANCE);
-
-        CraftingHelper.register(new ResourceLocation("forge", "compound"), CompoundIngredient.Serializer.INSTANCE);
-        CraftingHelper.register(new ResourceLocation("forge", "nbt"), NBTIngredient.Serializer.INSTANCE);
-        CraftingHelper.register(new ResourceLocation("forge", "partial_nbt"), PartialNBTIngredient.Serializer.INSTANCE);
-        CraftingHelper.register(new ResourceLocation("forge", "difference"), DifferenceIngredient.Serializer.INSTANCE);
-        CraftingHelper.register(new ResourceLocation("forge", "intersection"), IntersectionIngredient.Serializer.INSTANCE);
-        CraftingHelper.register(new ResourceLocation("minecraft", "item"), VanillaIngredientSerializer.INSTANCE);
-
-        event.getRegistry().register(new ConditionalRecipe.Serializer<Recipe<?>>().setRegistryName(new ResourceLocation("forge", "conditional")));
-
+        if (event.getRegistryKey().equals(ForgeRegistries.Keys.RECIPE_SERIALIZERS))
+        {
+            CraftingHelper.register(AndCondition.Serializer.INSTANCE);
+            CraftingHelper.register(FalseCondition.Serializer.INSTANCE);
+            CraftingHelper.register(ItemExistsCondition.Serializer.INSTANCE);
+            CraftingHelper.register(ModLoadedCondition.Serializer.INSTANCE);
+            CraftingHelper.register(NotCondition.Serializer.INSTANCE);
+            CraftingHelper.register(OrCondition.Serializer.INSTANCE);
+            CraftingHelper.register(TrueCondition.Serializer.INSTANCE);
+            CraftingHelper.register(TagEmptyCondition.Serializer.INSTANCE);
+
+            CraftingHelper.register(new ResourceLocation("forge", "compound"), CompoundIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "nbt"), NBTIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "partial_nbt"), PartialNBTIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "difference"), DifferenceIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("forge", "intersection"), IntersectionIngredient.Serializer.INSTANCE);
+            CraftingHelper.register(new ResourceLocation("minecraft", "item"), VanillaIngredientSerializer.INSTANCE);
```

Comment:
Perhaps `IConditionSerializer` and `IIngredientSerializer` could be made into proper (unsynced, unsaved) Forge registries, instead of being their own separate special registries in `CraftingHelper`.

Reply:
Nevermind this comment, I've been told that `IConditionSerializer` and `IIngredientSerializer` are to remain as their own special, lighter registries in `CraftingHelper`.